From f4d494ff1c89cc1ebad304823ccf806d872226c0 Mon Sep 17 00:00:00 2001
From: Yu Feng <rainwoodman@gmail.com>
Date: Sun, 23 Feb 2020 13:33:37 -0800
Subject: [PATCH 2/3] class-2.6.1-pathsize

---
 include/common.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/common.h b/include/common.h
index 06a97566..5157e4a7 100755
--- a/include/common.h
+++ b/include/common.h
@@ -27,7 +27,7 @@
 #define _ERRORMSGSIZE_ 2048 /**< generic error messages are cut beyond this number of characters */
 typedef char ErrorMsg[_ERRORMSGSIZE_]; /**< Generic error messages (there is such a field in each structure) */
 
-#define _FILENAMESIZE_ 256 /**< size of the string read in each line of the file (extra characters not taken into account) */
+#define _FILENAMESIZE_ 512 /**< size of the string read in each line of the file (extra characters not taken into account) */
 typedef char FileName[_FILENAMESIZE_];
 
 #define _PI_ 3.1415926535897932384626433832795e0 /**< The number pi */
-- 
2.20.1

